<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Access UKB data on Rosalind/CREATE • ukbkings</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Access UKB data on Rosalind/CREATE">
<meta property="og:description" content="ukbkings">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ukbkings</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.3.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/kcl-ukb-access.html">Access UKB data on Rosalind/CREATE</a>
    </li>
    <li>
      <a href="../articles/primary-care-prescriptions.html">Using UKB primary care prescription data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/kenhanscombe/ukbkings/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="kcl-ukb-access_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Access UKB data on Rosalind/CREATE</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/kenhanscombe/ukbkings/blob/HEAD/vignettes/kcl-ukb-access.Rmd" class="external-link"><code>vignettes/kcl-ukb-access.Rmd</code></a></small>
      <div class="hidden name"><code>kcl-ukb-access.Rmd</code></div>

    </div>

    
    
<hr>
<p><strong>Important. This document is only useful for UKB-approved KCL reasearchers and their collaborators, with an account on the Rosalind/CREATE HPC cluster.</strong></p>
<p>The ukbkings package works with a Rosalind/CREATE UKB project directory that has been setup to use the package. See <a href="https://github.com/kenhanscombe/ukbproject" class="external-link">ukbproject</a> for setup of the directory structure of a UKB project on Rosalind/CREATE.</p>
<hr>
<p><span style="color:dodgerblue;"><strong>Contents:</strong></span><br><a href="#tldr">0. tldr</a><br><a href="#installation">1. Installation</a><br><a href="#container">2. Container</a><br><a href="#field-subset-file">3. Field subset file</a><br><a href="#read-and-write-data">4. Read and write data</a><br><a href="#categorical-codes">5. Categorical codes</a><br><a href="#record-level-data">6. Record level data</a><br><a href="#genetic-data">7. Genetic data</a></p>
<hr>
<div class="section level3">
<h3 id="tldr">0. tldr<a class="anchor" aria-label="anchor" href="#tldr"></a>
</h3>
<p>Install the package</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"kenhanscombe/ukbkings"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span>, force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Or, use the docker container with ukbkings and dependencies installed</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1"></a><span class="ex">singularity</span> run docker://onekenken/ukbkings:0.2.2</span></code></pre></div>
<p>Write a serialised R dataframe to file for your required fields.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bio_phen.html">bio_phen</a></span><span class="op">(</span>
    project_dir <span class="op">=</span> <span class="st">"&lt;path_to_project_directory&gt;"</span>,
    field <span class="op">=</span> <span class="st">"&lt;path_to_required_fields_file&gt;"</span>, <span class="co"># one per line, no header</span>
    out <span class="op">=</span> <span class="st">"&lt;stem_of_output_file&gt;"</span> <span class="co"># e.g. "data/ukb" writes "data/ukb.rds"</span>
<span class="op">)</span></code></pre></div>
<p><em><strong>Note</strong>. <code>bio_phen</code> reads a withdrawal file from the project directory and replaces phenotype values for to-be-excluded samples with <code>NA</code>.</em></p>
<p><br></p>
<p>Read the generated dataset into R.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"data/ukb.rds"</span><span class="op">)</span></code></pre></div>
<p><br></p>
</div>
<div class="section level3">
<h3 id="installation">1. Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>Change to your user directory and make a study folder. Start an interactive cluster session with sufficient memory to read the UKB data.</p>
<p>The below procedure describes loading the HPC R module and installing ukbkings. Section 2 describes an easier alternative, which is running a containerized version of R with ukbkings and dependencies pre-installed.</p>
<p>Load the default cluster version of R. From your study folder in your user director on Rosalind:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1"></a><span class="ex">srun</span> -p shared,brc --mem=30G --pty /bin/bash</span>
<span id="cb5-2"><a href="#cb5-2"></a></span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="ex">module</span> avail 2<span class="op">&amp;&gt;</span>1 <span class="kw">|</span> <span class="fu">grep</span> R</span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="ex">module</span> load <span class="op">&lt;</span>default_R_from_above<span class="op">&gt;</span></span>
<span id="cb5-5"><a href="#cb5-5"></a></span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="ex">R</span></span></code></pre></div>
<p><br></p>
<p>Or, from your study folder in your user director on CREATE</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1"></a><span class="ex">srun</span> -p cpu --time=0-1 --mem=30G --pty /bin/bash</span>
<span id="cb6-2"><a href="#cb6-2"></a></span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="ex">module</span> -r spider <span class="st">'^r$'</span></span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="ex">module</span> load <span class="op">&lt;</span>default_R_from_above<span class="op">&gt;</span></span>
<span id="cb6-5"><a href="#cb6-5"></a></span>
<span id="cb6-6"><a href="#cb6-6"></a><span class="ex">R</span></span></code></pre></div>
<p>Install from Github</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"kenhanscombe/ukbkings"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span>, force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><br></p>
</div>
<div class="section level3">
<h3 id="container">2. Container<a class="anchor" aria-label="anchor" href="#container"></a>
</h3>
<p>Alternatively, from your interactive SLURM session, use a container with ukbkings and dependencies pre-installed</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># If on Rosalind load singularity; not needed on CREATE</span></span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="ex">module</span> load apps/singularity/3.5.3</span>
<span id="cb8-3"><a href="#cb8-3"></a></span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="ex">singularity</span> run docker://onekenken/ukbkings:0.2.2</span></code></pre></div>
<p><strong>Note</strong>: This must be done from within your study folder in your user directory as this will be the working directory in the containerized R session.</p>
<p><br></p>
<p><em><strong>Note</strong>. All code blocks below are R, unless otherwise specified.</em></p>
<p><br></p>
<p>Load libraries</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/kenhanscombe/ukbkings" class="external-link">ukbkings</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></code></pre></div>
<p>Check help (press ‘q’ to exit).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">?</span><span class="va">ukbkings</span></code></pre></div>
<p><em><strong>Note</strong>. All included data and functionality is described in the package index and also described on the <a href="https://kenhanscombe.github.io/ukbkings/" class="external-link">ukbkings webpage</a> under the <strong>Reference</strong> tab.</em></p>
<p><br></p>
<p>Check help on specific function, e.g.,</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">?</span><span class="va">bio_phen</span></code></pre></div>
<p>Point to the project directory.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">project_dir</span> <span class="op">&lt;-</span> <span class="st">"&lt;absolute_path_to_project_directory&gt;"</span></code></pre></div>
<p><br></p>
</div>
<div class="section level3">
<h3 id="field-subset-file">3. Field subset file<a class="anchor" aria-label="anchor" href="#field-subset-file"></a>
</h3>
<p>You need a file with required fields, one per line, no header.</p>
<p><br></p>
<p>Read the project field-to-name “field finder” file, inspect the variable metadata, and display the number of baskets included.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bio_field.html">bio_field</a></span><span class="op">(</span><span class="va">project_dir</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span>
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span>

<span class="va">f</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">basket</span><span class="op">)</span></code></pre></div>
<p>Search for variables required and add their field codes to a file, one per line, no header. You can page through the file.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/utils/page.html" class="external-link">page</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"print"</span><span class="op">)</span></code></pre></div>
<p>Or, search <code>name</code> column</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">field</span>, <span class="va">name</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">name</span>, <span class="st">"vegetables"</span><span class="op">)</span><span class="op">)</span>

<span class="va">f</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">field</span>, <span class="va">name</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">name</span>, <span class="st">"ldl|triglycerides"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Alternatively, search the <a href="http://biobank.ndph.ox.ac.uk/showcase/" class="external-link">UKB showcase</a> for a variable of interest then filter on the <code>field</code> column in the field-to-name dataframe (useful if multiple instances required). For example, if you search for “cholesterol medication”, the field stem you want is 6177.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">field</span>, <span class="va">name</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">field</span>, <span class="st">"6177"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><code>bio_field_add</code> is a convenience function for creating the one per line required variables/fields file. By default the function appends fields. Create the field subset file in your UKB user directory</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">field</span>, <span class="va">name</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">field</span>, <span class="st">"6177"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="../reference/bio_field_add.html">bio_field_add</a></span><span class="op">(</span><span class="st">"small_field_subset.txt"</span><span class="op">)</span></code></pre></div>
<p>Inspect the field selection file.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/system.html" class="external-link">system</a></span><span class="op">(</span><span class="st">"cat small_field_subset.txt"</span><span class="op">)</span></code></pre></div>
<p><br></p>
</div>
<div class="section level3">
<h3 id="read-and-write-data">4. Read and write data<a class="anchor" aria-label="anchor" href="#read-and-write-data"></a>
</h3>
<p>Read required fields and save as an rds file in your user directory. Argument <code>out</code> should be a path to your UKB user directory</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bio_phen.html">bio_phen</a></span><span class="op">(</span>
    <span class="va">project_dir</span>,
    field <span class="op">=</span> <span class="st">"small_field_subset.txt"</span>,
    out <span class="op">=</span> <span class="st">"small_phenotype_subset"</span>
<span class="op">)</span></code></pre></div>
<p><em><strong>Note</strong>. Dates in the UKB data are recorded in a variety of formats, some of which are non-standard: “character string is not in a standard unambiguous format”, e.g., 2009-01-12T11:28:56. All date variables have been left in character format for the user to convert as needed.</em></p>
<p><br></p>
<p>Check the size of your file and read in your dataset</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/system.html" class="external-link">system</a></span><span class="op">(</span><span class="st">"ls -lh small_phenotype_subset.rds"</span><span class="op">)</span>
<span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"small_phenotype_subset.rds"</span><span class="op">)</span></code></pre></div>
<p>If required, rename columns from the default UKB field names to the descriptive names used in the field-to-name “field finder” <code>name</code> column.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bio_rename.html">bio_rename</a></span><span class="op">(</span><span class="va">df</span>, <span class="va">f</span><span class="op">)</span></code></pre></div>
<p><br></p>
</div>
<div class="section level3">
<h3 id="categorical-codes">5. Categorical codes<a class="anchor" aria-label="anchor" href="#categorical-codes"></a>
</h3>
<p>Categorical field codings are included in the field finder.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">field</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">field</span>, <span class="va">categorical_coding</span><span class="op">)</span></code></pre></div>
<p>Retrieve numerical “Value” and and associated “Meaning” for each categorical code.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cx</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bio_code.html">bio_code</a></span><span class="op">(</span><span class="va">project_dir</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cx</span><span class="op">)</span></code></pre></div>
<p>Look up a particular coding.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cx</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Coding</span> <span class="op">==</span> <span class="fl">502</span><span class="op">)</span></code></pre></div>
<p><br></p>
</div>
<div class="section level3">
<h3 id="record-level-data">6. Record level data<a class="anchor" aria-label="anchor" href="#record-level-data"></a>
</h3>
<p>To query and read record-level data, use <code>bio_record</code> and <code>bio_record_map</code>. List all record level data available for your project</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bio_record.html">bio_record</a></span><span class="op">(</span><span class="va">project_dir</span><span class="op">)</span></code></pre></div>
<p><br></p>
<p>You can retrieve the data as a <a href="https://diskframe.com/index.html" class="external-link">disk.frame</a> which you can inspect with functions like <code>head</code>, <code>names</code>, etc. This data is still on disk and so does not require a large amount of memory to read into R.</p>
<p>You can also pipe the disk.frame through dplyr verbs, e.g., for column selection and row filtering. Use <code>collect</code> to read the results from disk into an R dataframe. For example to retieve death data</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gp_scripts_diskf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bio_record.html">bio_record</a></span><span class="op">(</span><span class="va">project_dir</span>, record <span class="op">=</span> <span class="st">"gp_scripts"</span><span class="op">)</span>

<span class="va">gp_scripts_df</span> <span class="op">&lt;-</span> <span class="va">gp_scripts_diskf</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">drug_name</span>, <span class="st">"fluoxetine"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">eid</span>, <span class="va">data_provider</span>, <span class="va">issue_date</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/compute.html" class="external-link">collect</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Use the <code>subset</code> argument as a convenient way to read only the data for the samples you’re interested in, e.g. those with data on your phenotype of interest. This will automatically read the data into a dataframe in R - you do not need to <code>collect</code>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sample_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">321</span>, <span class="fl">654</span>, <span class="fl">987</span><span class="op">)</span>

<span class="va">gp_scripts_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bio_record.html">bio_record</a></span><span class="op">(</span>
    <span class="va">project_dir</span>,
    record <span class="op">=</span> <span class="st">"gp_scripts"</span>,
    subset <span class="op">=</span> <span class="va">sample_subset</span>
<span class="op">)</span></code></pre></div>
<p><br></p>
<p>To inspect several records at once use <code>bio_record_map</code>, which maps a function to the data on disk without reading it into R. For example, to quickly find which variables are in which tables</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bio_record_map.html">bio_record_map</a></span><span class="op">(</span><span class="va">project_dir</span>, func <span class="op">=</span> <span class="va">names</span><span class="op">)</span></code></pre></div>
<p>By default the function is mapped to all record table. You can also specify the records you’re interested in</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bio_record_map.html">bio_record_map</a></span><span class="op">(</span>
    <span class="va">project_dir</span>,
    func <span class="op">=</span> <span class="va">head</span>,
    records <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gp_clinical"</span>, <span class="st">"gp_scripts"</span>, <span class="st">"gp_registrations"</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p><br></p>
<p>For GP record details see UKB documentation:</p>
<ul>
<li><a href="http://biobank.ndph.ox.ac.uk/showcase/label.cgi?id=3001" class="external-link">Category 3001</a></li>
<li><a href="http://biobank.ndph.ox.ac.uk/showcase/refer.cgi?id=591" class="external-link">Resource 591: Primary Care Linked Data</a></li>
<li><a href="http://biobank.ndph.ox.ac.uk/showcase/refer.cgi?id=592" class="external-link">Resource 592: Clinical coding classification systems and maps</a></li>
</ul>
<p><br></p>
<p>For COVID-19 record details see UKB documentation:</p>
<ul>
<li><a href="http://biobank.ndph.ox.ac.uk/showcase/field.cgi?id=40100" class="external-link">Data field 40100</a></li>
<li><a href="http://biobank.ndph.ox.ac.uk/showcase/exinfo.cgi?src=COVID19_tests" class="external-link">COVID-19 test results data</a></li>
<li><a href="https://biobank.ndph.ox.ac.uk/ukb/refer.cgi?id=3151" class="external-link">Resource 3151: Primary Care (GP) Data for COVID-19 Research</a></li>
</ul>
<p><br></p>
<p>For HES in-patient record details see UKB documentation:</p>
<ul>
<li><a href="https://biobank.ndph.ox.ac.uk/showcase/label.cgi?id=2000" class="external-link">Category 2000</a></li>
<li><a href="https://biobank.ndph.ox.ac.uk/showcase/label.cgi?id=2006" class="external-link">Category 2006</a></li>
<li><a href="https://biobank.ndph.ox.ac.uk/showcase/refer.cgi?id=138483" class="external-link">Resource 138483: Hospital Episode Statistics</a></li>
<li><a href="https://biobank.ndph.ox.ac.uk/showcase/refer.cgi?id=141140" class="external-link">Resource 141140: HES Data Dictionary</a></li>
</ul>
<p><br></p>
<p>For death record details see UKB documentation:</p>
<ul>
<li><a href="https://biobank.ctsu.ox.ac.uk/crystal/label.cgi?id=100093" class="external-link">Category 100093</a></li>
<li><a href="https://biobank.ctsu.ox.ac.uk/crystal/refer.cgi?id=134993" class="external-link">Resource 134993: Health Outcomes Report</a></li>
</ul>
<p><br></p>
</div>
<div class="section level3">
<h3 id="genetic-data">7. Genetic data<a class="anchor" aria-label="anchor" href="#genetic-data"></a>
</h3>
<p>For paths to the genetic data available for the project</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bio_gen_ls.html">bio_gen_ls</a></span><span class="op">(</span><span class="va">project_dir</span><span class="op">)</span></code></pre></div>
<p>For the genotyped data “sample information” files fam file and sample QC files</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bio_gen_fam.html">bio_gen_fam</a></span><span class="op">(</span><span class="va">project_dir</span><span class="op">)</span>
<span class="fu"><a href="../reference/bio_gen_sqc.html">bio_gen_sqc</a></span><span class="op">(</span><span class="va">project_dir</span><span class="op">)</span></code></pre></div>
<p>Read the relatedness data into a dataframe with <code>bio_gen_related</code>. To get a dataframe of related samples to remove use <code>bio_gen_related_remove</code> - uses <a href="https://gitlab.com/choishingwan/GreedyRelated" class="external-link">GreedyRelated</a> with default relatedness theshold, <code>thresh = 0.044</code></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bio_gen_related.html">bio_gen_related</a></span><span class="op">(</span><span class="va">project_dir</span><span class="op">)</span>
<span class="fu"><a href="../reference/bio_gen_related_remove.html">bio_gen_related_remove</a></span><span class="op">(</span><span class="va">project_dir</span><span class="op">)</span></code></pre></div>
<p>To assign 1000 Genomes super population ancestry to your project-specific pseudo-IDs.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bio_gen_ancestry.html">bio_gen_ancestry</a></span><span class="op">(</span><span class="va">project_dir</span><span class="op">)</span></code></pre></div>
<p>To write a PLINK file of samples to <code>keep</code>, use <code>bio_gen_write_plink_input</code>: <code>data</code> is either a vector of samples to keep, or a dataframe with samples IDs to keep in the first column; <code>out</code> is the file path to write output file to.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bio_gen_write_plink_input.html">bio_gen_write_plink_input</a></span><span class="op">(</span><span class="va">data</span>, <span class="va">out</span><span class="op">)</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Ken Hanscombe.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
